odo/__init__.py

```diff
@@ -1,73 +1,37 @@
-from __future__ import absolute_import, division, print_function
+from __future__ import absolute_import as _
+from __future__ import division as _
+from __future__ import print_function as _
+
+from pkg_resources import DistributionNotFound as _DistributionNotFound
+from pkg_resources import get_distribution as _get_distribution
+
+try:
+    __version__ = _get_distribution('odo').version
+except _DistributionNotFound:
+    __version__ = 'unknown'
 
 try:
-    import h5py  #  h5py has precedence over pytables
-except:
+    # h5py has precedence over pytables
+    import h5py as _
+except ImportError:
     pass
-from multipledispatch import halt_ordering, restart_ordering
 
-halt_ordering()  # Turn off multipledispatch ordering
+from multipledispatch import halt_ordering as _halt_ordering
+from multipledispatch import restart_ordering as _restart_ordering
+
+# Turn off multipledispatch ordering
+_halt_ordering()
 
-from .utils import ignoring
 from .convert import convert
 from .append import append
 from .resource import resource
-from .directory import Directory
 from .into import into
 from .odo import odo
 from .create import create
 from .drop import drop
-from .temp import Temp
-from .backends.text import TextFile
-from .chunks import chunks, Chunks
 from datashape import discover, dshape
-import numpy as np
-
-with ignoring(ImportError):
-    from .backends.sas import sas7bdat
-with ignoring(ImportError):
-    from .backends.pandas import pd
-with ignoring(ImportError):
-    from .backends.bcolz import bcolz
-with ignoring(ImportError):
-    from .backends.h5py import h5py
-with ignoring(ImportError):
-    from .backends.hdfstore import HDFStore
-with ignoring(ImportError):
-    from .backends.pytables import PyTables
-with ignoring(ImportError):
-    from .backends import sql
-with ignoring(ImportError):
-    from .backends import mongo
-with ignoring(ImportError):
-    from .backends.csv import CSV
```

Comment:
I think all of the types we define should still be part of the top level API, thinks like `odo.CSV` are very useful and we shouldn't break code that currently depends on those names being there.